apps/emqx_bridge_pulsar/src/emqx_bridge_pulsar_impl_producer.erl

```diff
@@ -422,3 +431,19 @@ partition_strategy(Strategy) -> Strategy.
 
 is_sensitive_key(auth_data) -> true;
 is_sensitive_key(_) -> false.
+
+get_error_message({BrokerErrorMap, _}) when is_map(BrokerErrorMap) ->
+    Iter = maps:iterator(BrokerErrorMap),
+    do_get_error_message(Iter);
+get_error_message(_Error) ->
+    error.
+
+do_get_error_message(Iter) ->
```

Comment:
Q: Why filter error messages

Reply:
The error value from Pulsar usually contains a more user-friendly message detailing what went wrong, as exemplified in the screenshot in the PR description.  If that info is available, we try to present that to the user; otherwise, we throw a generic error.